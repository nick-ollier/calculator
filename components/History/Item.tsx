import { formatHistoryExpression } from '@utils';

export interface ItemProps {
    value: string;
    expression: string;
}
const Item: React.FC<ItemProps> = ({ value, expression }) => (
    <div
        role='presentation'
        aria-label={`${expression}=${value}`}
        className='mb-4 text-right cursor-pointer'
        onClick={() => navigator.clipboard.writeText(value)}
    >
        <p className='text-sm'>{formatHistoryExpression(expression)}</p>
        <p className='text-2xl font-bold'>{value}</p>
    </div>
);

export default Item;
