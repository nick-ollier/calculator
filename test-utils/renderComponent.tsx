import CalculatorProvider from '@context/CalculatorContext';
import { render } from '@testing-library/react';

const renderWrapper = (ui: React.ReactElement) => {
    const WithProviders = ({ children }) => {
        return <CalculatorProvider>{children}</CalculatorProvider>;
    };

    return render(ui, { wrapper: WithProviders });
};

export * from '@testing-library/react';
export { renderWrapper as render };
