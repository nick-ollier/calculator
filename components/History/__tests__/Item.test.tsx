import { render, screen } from '@testUtils/renderComponent';
import userEvent from '@testing-library/user-event';
import Item, { ItemProps } from '@components/History/Item';

const renderComponent = (props?: Partial<ItemProps>) => {
    const utils = render(<Item value='2' expression='1+1' {...props} />);

    return {
        ...utils
    };
};

describe('component: history item', () => {
    it('renders', () => {
        const { container } = renderComponent();
        expect(container);
    });

    it('matches snapshot', () => {
        const { asFragment } = renderComponent();
        expect(asFragment()).toMatchSnapshot();
    });

    it('should copy value to clipboard on click of expression/value', () => {
        Object.assign(navigator, {
            clipboard: {
                writeText: () => null
            }
        });

        jest.spyOn(navigator.clipboard, 'writeText');

        renderComponent();

        const sum = screen.getByRole('presentation', {
            name: /1\+1=2/i
        });

        userEvent.click(sum);
        expect(navigator.clipboard.writeText).toHaveBeenCalledWith('2');
    });
});
