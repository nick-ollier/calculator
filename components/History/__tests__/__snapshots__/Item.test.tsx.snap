// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: history item matches snapshot 1`] = `
<DocumentFragment>
  <div
    aria-label="1+1=2"
    class="mb-4 text-right cursor-pointer"
    role="presentation"
  >
    <p
      class="text-sm"
    >
      1 + 1 =
    </p>
    <p
      class="text-2xl font-bold"
    >
      2
    </p>
  </div>
</DocumentFragment>
`;
